src/v2/guide/composition.md

```diff
@@ -0,0 +1,529 @@
+  <span slot="bold">Error!</span>
+  Something went wrong.
+</app-note>
+```
+
+But there's a problem. Every time you display an error, you'll have to remember to add `type="danger"` and `Error!` in the `bold` slot. It's easy to make a mistake, resulting in inconsistent error messages.
+
+Ideally, we'd avoid error-prone repetition with an `ErrorNote` component, where we'd simply provide the error text:
+
+``` html
+<error-note>
+  Something went wrong.
+</error-note>
+```
+
+Then this component would automatically prefix the content with "Error!" in bold and add the correct class.
+
+### The Problems with Copying Component Definitions
+
+We could create this new `ErrorNote` component by copying and pasting the definition for `AppNote`, then making a few changes:
+
+``` js
+var ErrorNote = {
+  template: `
+    <p class="note danger">
+      <strong>
+        Error!
+      </strong>
+      <slot/>
+    </p>
+  `
+}
+```
+
+This is the simplest option and sometimes it's appropriate, when two components only _incidentally_ share a lot in common. In this case, however, we probably want all the notes in our application to work mostly the same. For example, if we later added a button to expand/hide details to `AppNote`, we'd now have to add the exact same code to `ErrorNote` and any other variations we created.
+
+Fortunately, there's a better way.
+
+### Merging Options with `extends`
+
+In cases like this, the `extends` option is usually a more appropriate alternative:
+
+``` js
+var ErrorNote = {
+  extends: AppNote,
+  props: {
+    type: {
+      type: String,
+      default: 'danger'
+    }
+  },
+  beforeCreate: function () {
+    if (!this.$slots.bold) {
+      this.$slots.bold = this.$createElement('span', 'Error!')
+    }
+  }
+}
+```
+
+In the above example, `extends` is telling Vue to merge our config with the `AppNote` config. In all cases, the component doing the extension (`ErrorNote`) takes priority, so that:
+
+- The definition for the `type` prop from `AppNote` is replaced with a new definition with a default of `'danger'`.
+- A `beforeCreate` hook is added to create a default of `<span>Error!</span>` for the `bold` slot.
+
+### Merging of Instance Properties
+
+For options that create instance properties, such as `props`, `data`, `computed`, and `methods`, Vue's merging does **not** replace the entire definition. For example, in this case:
+
+``` js
+var ComponentA = {
+  props: {
+    foo: String,
+    bar: Number
+  }
+}
+
+var ComponentB = {
+  extends: ComponentA,
+  props: {
+    foo: {
+      type: String,
+      default: 'hi'
+    },
+    baz: Boolean
+  }
+}
+```
+
+The resulting options object from the `extends` in `ComponentB` would be:
+
+``` js
+{
+  props: {
+    foo: {
+      type: String,
+      default: 'hi'
+    },
+    bar: Number,
+    baz: Boolean
+  }
+}
+```
+
+Nothing is removed, prop definitions with the same name are replaced, and any new props are added.
+
+### Merging of Event Handlers
```

Comment:
I think it's more accurate to just use "Merging of Lifecycle Hooks" here. Event handlers may be confused with handlers added via `this.$on`.

Reply:
Good point!